src/agent/io/scanner.ts

```diff
@@ -223,29 +230,40 @@ function findFiles(baseDir: string, regex: RegExp): Promise<string[]> {
 function statsForFile(filename: string,
                       shouldHash: boolean): Promise<FileStats> {
   return new Promise<FileStats>((resolve, reject) => {
-    let shasum: crypto.Hash;
-    if (shouldHash) {
-      shasum = crypto.createHash('sha1');
-    }
-    // TODO: Determine why property 'ReadStream' does not exist on type 'fs'
-    const s = (fs as {} as {ReadStream: Function}).ReadStream(filename);
-    let lines = 0;
-    const byLine = s.pipe(split());
-    byLine.on('error', (e: Error) => {
-      reject(e);
-    });
-    byLine.on('data', (d: string) => {
-      if (shouldHash) {
-        shasum.update(d);
-      }
-      lines++;
+    const reader = fs.createReadStream(filename);
+    reader.on('error', (err) => {
+      reject(err);
     });
-    byLine.on('end', () => {
-      let d: string|undefined;
+    reader.on('open', () => {
+      let shasum: crypto.Hash;
       if (shouldHash) {
-        d = shasum.digest('hex');
+        shasum = crypto.createHash('sha1');
       }
-      resolve({hash: d, lines});
+
+      let lines = 0;
+      let error: Error|undefined;
+      const byLine = reader!.pipe(split());
+      byLine.on('error', (e: Error) => {
+        error = e;
+      });
+      byLine.on('data', (d: string) => {
+        if (shouldHash) {
+          shasum.update(d);
+        }
+        lines++;
+      });
+      byLine.on('end', () => {
+        if (error) {
+          reject(error);
+        }
+        else {
+          let d: string|undefined;
```

Comment:
I've updated it to
```js
const hash = shouldHash ? shasum.digest('hex') : undefined;
resolve({hash, lines});
```
This is needed because
```js
shouldHash && 'some text'
```
is of type `string` if `shouldHash` is `true` and of type `boolean` if `shouldHash` is `false`.  However, `hash` is expected to be of type `string|undefined`.